superset/tasks/schedules.py

```diff
@@ -645,55 +660,65 @@ def deliver_slack_alert(alert_content: AlertContent, slack_channel: str) -> None
     )
 
 
-def run_alert_query(
-    alert_id: int, database_id: int, sql: str, label: str
-) -> Optional[bool]:
+def observe(alert_id: int) -> str:
     """
-    Execute alert.sql and return value if any rows are returned
+    Runs the SQL query in an alert's SQLObserver and then
+    stores the result in a SQLObservation
     """
-    logger.info("Processing alert ID: %i", alert_id)
-    database = db.session.query(Database).get(database_id)
-    if not database:
-        logger.error("Alert database not preset")
-        return None
 
-    if not sql:
-        logger.error("Alert SQL not preset")
-        return None
+    sql_observer = db.session.query(SQLObserver).filter_by(alert_id=alert_id).one()
+    value = None
 
-    parsed_query = ParsedQuery(sql)
+    parsed_query = ParsedQuery(sql_observer.sql)
     sql = parsed_query.stripped()
+    df = sql_observer.database.get_df(sql)
+
+    if not df.empty:
+        value = float(df.to_records()[0][1])
```

Comment:
add logging here if there are incorrect types / df is empty and properly handle edgecases
for instance you can add a type / field to the observation that result was malformed